js/dom_types.ts

```diff
@@ -536,3 +536,73 @@ export interface Response extends Body {
   /** Creates a clone of a `Response` object. */
   clone(): Response;
 }
+
+export interface Location {
+  /**
+   * Returns a DOMStringList object listing the origins of the ancestor browsing
+   * contexts, from the parent browsing context to the top-level browsing
+   * context.
+   */
+  readonly ancestorOrigins: string[];
+  /**
+   * Returns the Location object's URL's fragment (includes leading "#" if
+   * non-empty).
+   * Can be set, to navigate to the same URL with a changed fragment (ignores
+   * leading "#").
+   */
+  hash: string;
+  /**
+   * Returns the Location object's URL's host and port (if different from the
+   * default port for the scheme).  Can be set, to navigate to the same URL with
+   * a changed host and port.
+   */
+  host: string;
+  /**
+   * Returns the Location object's URL's host.  Can be set, to navigate to the
+   * same URL with a changed host.
+   */
+  hostname: string;
+  /**
+   * Returns the Location object's URL.  Can be set, to navigate to the given
+   * URL.
+   */
+  href: string;
+  /** Returns the Location object's URL's origin. */
+  readonly origin: string;
+  /**
+   * Returns the Location object's URL's path.
+   * Can be set, to navigate to the same URL with a changed path.
+   */
+  pathname: string;
```

Comment:
Since this property is not read-only, what happens when I set them?

Reply:
It was copy and pasted from TS's declaration. In the browser you can set these properties. In Deno it doesn't make sense. I've changed them to be readonly.

Reply:
Actually. On second thought, I don't think we should deviate from the browser TS's declaration so that we don't make isomorphic location code difficult. 

In Deno assignment to these various properties will be a no-op for now and potentially we should throw in the future. But I'll leave that for future work.